File: Flink-Connectors/flink-redis/src/main/scala/com/ng/flink/RedisExampleMapper.scala
package com.ng.flink

import org.apache.flink.streaming.connectors.redis.RedisSink
import org.apache.flink.streaming.connectors.redis.common.config.{FlinkJedisPoolConfig, FlinkJedisSentinelConfig}
import org.apache.flink.streaming.connectors.redis.common.mapper.{RedisCommand, RedisCommandDescription, RedisMapper}

/**
  * @Author: Cedaris
  * @Date: 2019/6/25 10:24
  */
class RedisExampleMapper1 extends RedisMapper[(String,String)]{
  override def getCommandDescription: RedisCommandDescription = {
    new RedisCommandDescription(RedisCommand.HSET,"HASH_NAME")
  }

  override def getKeyFromData(data: (String, String)): String = {
    data._1
  }

  override def getValueFromData(data: (String, String)): String = {
    data._2
  }

  def main(args: Array[String]): Unit = {
    //single redis server:
    val conf: FlinkJedisPoolConfig = new FlinkJedisPoolConfig.Builder().setHost("217.0.0.0").build()

//    stream.addSink(new RedisSink[(String, String)](conf, new RedisExampleMapper))

//for Redis Cluster
  /*val conf = new FlinkJedisSentinelConfig.Builder().setMasterName("master")
   .setSentinels(...).build()
    stream.addSink(new RedisSink[(String, String)](conf, new RedisExampleMapper))*/
  }
}

File: Flink-Connectors/flink-redis/src/main/java/com/ng/flink/RedisExampleMapper.java
package com.ng.flink;

import org.apache.flink.streaming.connectors.redis.common.mapper.RedisCommand;
import org.apache.flink.streaming.connectors.redis.common.mapper.RedisCommandDescription;
import org.apache.flink.streaming.connectors.redis.common.mapper.RedisMapper;
import scala.Tuple2;

/**
 * how to create a sink that communicate to a single redis server
 * @Author: Cedaris
 * @Date: 2019/6/25 10:10
 */
public class RedisExampleMapper implements RedisMapper<Tuple2<String,String>> {

    @Override
    public RedisCommandDescription getCommandDescription() {
        return new RedisCommandDescription(RedisCommand.HSET, "HASH_NAME");
    }

    @Override
    public String getKeyFromData(Tuple2<String, String> data) {
        return data._1;
    }

    @Override
    public String getValueFromData(Tuple2<String, String> data) {
        return data._2;
    }
}
